components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -112,4 +117,35 @@ function AutomaticPointsTable:parseManualPoints(team, tournamentCount)
 	return manualPoints
 end
 
+function AutomaticPointsTable:queryPlacements(tournaments)
+	local queryParams = {
+		limit = 5000,
+		query = 'participant, placement, extradata'
+	}
+
+	local tournamentsWithPlacements = Table.mapValues(tournaments,
+		function(tournament)
+			local placements = {}
+
+			queryParams.conditions = '[[tournament::' .. tournament.name .. ']]'
+			local tournamentResults = mw.ext.LiquipediaDB.lpdb('placement', queryParams)
+
+			for _, result in pairs(tournamentResults) do
+				result.points = tonumber(result.extradata.prizepoints)
+				result.securedPoints = tonumber(result.extradata.securedpoints)
+				result.extradata = nil
+
+				local teamName = result.participant
+				placements[teamName] = result
+			end
+
+			local tournamentWithPlacements = Table.deepCopy(tournament)
+			tournamentWithPlacements.placements = placements
+			return tournamentWithPlacements

```

Comment:
Why do a deepcopy of each tournament instead of updating the tournament with the placements?

Reply:
I've been turning all functions to pure functions, which shouldn't alter the input, not sure if it's a requirement but I've noticed Vogan pointing it out

To be fair it's easier to change things in-place, not sure what I should be committing to now though..

Reply:
It's a good instinct but I think the starting off point for this function is wrong. You're essentially painting yourself into the corner of having to make that choice by using the `map` on `tournaments`. I would instead recommend a for loop where for each tournament you set the placements.